e2e/tests/ui-driven/src/helpers/addComponent.ts

```diff
@@ -44,20 +62,68 @@ const createBaseInput = async (
       await page.locator("id=undefined-max-month").fill("12");
       await page.locator("id=undefined-max-year").fill("2199");
       break;
-    case "Address Input":
+    case PlanXEditorComponent.ADDRESS:
       await page.getByPlaceholder("Title").fill(title || "");
       await page.getByPlaceholder("Data Field").fill(options?.[0] || "");
       break;
-    case "Contact Input":
+    case PlanXEditorComponent.CONTACT:
       await page.getByPlaceholder("Title").fill(title || "");
       await page.getByPlaceholder("Data Field").fill(options?.[0] || "");
+      break;
+    case PlanXEditorComponent.TASKLIST:
+      await page.getByPlaceholder("Main Title").fill(title || "");
+      if (options) {
+        let index = 0;
+        for (const option of options) {
+          await page.locator("button").filter({ hasText: "add new" }).click();
+          await page
+            .getByPlaceholder("Title")
+            .nth(index + 1) // ignore the main title field
+            .fill(option);
+          index++;
+        }
+      }
+      break;
+    case PlanXEditorComponent.REVIEW:
+      // Don't need to change anything so dummy click
+      await page
+        .getByPlaceholder("Check your answers before sending your application")
+        .click();
+      break;
+    case PlanXEditorComponent.FIND_PROPERTY:
+      break;
+    case PlanXEditorComponent.PLANNING_CONSTRAINTS:
+      break;
+    case PlanXEditorComponent.DRAW_BOUNDARY:
+      break;
+    case PlanXEditorComponent.NEXT_STEPS:
+      if (options) {
+        let index = 0;
+        for (const option of options) {
+          await page.locator("button").filter({ hasText: "add new" }).click();
+          await page
+            .getByPlaceholder("Title")
+            .nth(index + 1) // ignore the main title field
+            .fill(option);
+          index++;
+        }
+      }
+      break;
+    case PlanXEditorComponent.FILE_UPLOAD:
+      await page.getByPlaceholder("Data Field").fill(options?.[0] || "");
+
       break;
     default:
       throw new Error(`Unsupported type: ${type}`);
   }
 
   // convert type name to lowercase, with dashes if there are spaces
-  const buttonName = type.toLowerCase().replace(/\s/g, "-");
+  // or custom name if it doesn't fit the pattern
+  const buttonName =
+    type === PlanXEditorComponent.FIND_PROPERTY
+      ? "find-property-merged"
+      : type.toLowerCase().replace(/\s/g, "-");
+
   await page
```

Comment:
This seems like a bit of complexity we can remove - the button has `type="submit"` which we can access via a Playwright locator, or we can apply a `data-testid` to the button.